main.py

```diff
@@ -1,53 +1,223 @@
+            loss_sum += loss.item()
+            _, preds = torch.max(out, 1)
+            for pred, label in zip(preds, labels):
+                if pred == label:
+                    correct += 1
+    val_loss = loss_sum / len(test_dataloader)
+    val_accuracy = correct / total
+    return val_loss, val_accuracy
+
+
+def test_model(model: nn.Module, test_data: DataLoader, path: Optional[str] = None, classnames=None):
+    """
+    Evaluate the performance of the trained ComposerClassifier model with the provided test data loader.
+
+    Args:
+        model (PitchSeqNN): The trained ComposerClassifier model to be evaluated.
+        test_data (DataLoader): Data for the model to be tested on
+        path (str, optional): If specified, the function will load the model's state from the
+                              provided path.
+        classnames (list[str], optional): Classnames for printing results.
+    """
+    if classnames is None:
+        classnames = [0, 1]
+    # if path is specified load state from file
+    if path is not None:
+        model.load_state_dict(torch.load(path))
+    # containers for counting prediction
+    correct_pred = {classname: 0 for classname in classnames}
+    total_pred = {classname: 0 for classname in classnames}
+    correct = 0
+    total = 0
+    with torch.no_grad():
+        for data in test_data:
+            notes, labels = data
+            out = model(notes)
+            loss = nn.CrossEntropyLoss()(out, labels)
+            print(loss)
+            _, preds = torch.max(out, 1)
+            correct += (preds == labels).sum().item()
+            total += labels.size(0)
+
+            for label, pred in zip(labels, preds):
+                if label == pred:
+                    correct_pred[classnames[label]] += 1
+                total_pred[classnames[label]] += 1
+    for classname, correct_count in correct_pred.items():
+        accuracy = 100 * float(correct_count) / total_pred[classname]
+        print(f"Accuracy for class: {classname:5s} is {accuracy:.1f} %")
+    print(f"Accuracy of the network on the test data: {(100 * correct / total):.3f} %")
+    print(f"correctly predicted : {correct_pred}")
+    return correct / total
+
+
+def get_dataloader(split="train", sequence_length=64, batch_size=BATCH_SIZE):
+    """Create and prepare a DataLoader from ComposerClassificationDataset.
+
+    Parameters:
+        split (str): dataset split. Possible splits: "train", "test", "validation"
+        sequence_length (int): Length of a note sequence to be extracted.
+        batch_size (int): A batch size for DataLoader.
+    """
+    data = ComposerClassificationDataset(split=split, sequence_length=sequence_length)
+    data.samples = prepare_samples_pitch_only(data)
+    dataloader = DataLoader(data, batch_size=batch_size, shuffle=True)
+    return dataloader
+
+
+def main():
+    models = [PitchSeqNN(), PitchSeqNNv2_64(), PitchSeqNNv3_64(), PitchSeqNNv4_64()]
```

Comment:
```suggestion
    layers = [
      [64, 128, 256, 64, 2],
      [64, 128, 64, 2],
      ...
    ]
    for layer in layers:
      model = PitchNN(layers=layers)
```

As far as I can tell all versions of this model differ only in the number and shape of linear layers. I think it's a better approach to have a single model class and treat those values as parameters, especially if you want to check results for multiple configurations like this.